packages/kbn-ebt-tools/src/performance_metric_events/helpers.ts

```diff
@@ -34,5 +34,13 @@ export function reportPerformanceMetricEvent(
   analytics: Pick<AnalyticsClient, 'reportEvent'>,
   eventData: PerformanceMetricEvent
 ) {
+  if (eventData.eventName === 'kibana_started') {
+    // eslint-disable-next-line no-console
+    console.warn(
+      `KIBANA_STARTED is reported at ${new Date().toISOString()}, build id ${
+        process.env.BUILDKITE_BUILD_ID
+      }`
+    );
```

Comment:
This will log **in production** a **warning** claiming `KIBANA_STARTED is reported at {DATE}, build id undefined`.

IMO, we should not log warnings to users about events being sent. Because it's not an error, nor misbehavior they can interact with.

Also, I want to highlight that the event is not reported (sent by the shippers) as soon as `analytics.reportEvent` is called. `kibana_started` is a server-side event, and the batching mechanism is more aggressive, as it could take up to 10 minutes:

https://github.com/elastic/kibana/blob/0824234fe1da884221ddd1f42d4e5d072b7ee2f3/packages/analytics/shippers/elastic_v3/server/src/server_shipper.ts#L229-L242

Logging when the event was generated is not a guarantee that it was sent. Is that what we are aiming to measure? May I suggest we get this data by looking at the event's timestamp (generated as soon as `reportEvent` is called) and the `timestamp` attached by the receiving side (generated when the receiving end receives the request). cc @elastic/platform-analytics to keep me honest here.

There is this request (https://github.com/elastic/kibana/issues/140521), opened by @spalger to force-flush the events instead of depending on the batching logic so we can finish our tests earlier. While that issue is tackled, if you want to wait for the events to be sent, you may subscribe to `analytics.telemetryCounters$` and react to the `succeeded`/`failed`/`dropped` counter types.

Alternative, if you want to rely on logs, we can leverage this debug log that logs all the generated events:
https://github.com/elastic/kibana/blob/af45ef831ebc5e21586342a98896d6ab03563e0b/packages/analytics/client/src/analytics_client/analytics_client.ts#L136-L141

We can discuss removing the `if (isDev)` check and always debug-log it.

Reply:
I think what we are looking is exactly what you mentioned @afharo I didn't want to change the `isDev` part but will be great if we can see those logs at least in another branch that we can run journeys on. I'm going to remove that condition and create another target branch in order to see these logs.